astropy/utils/data.py

```diff
@@ -1375,15 +1714,103 @@ def get_cached_urls():
     -------
     cached_urls : list
         List of cached URLs.
-    """
-    # The code below is modified from astropy.utils.data.download_file()
-    try:
-        dldir, urlmapfn = _get_download_cache_locs()
-    except OSError as e:
-        msg = 'Remote data cache could not be accessed due to '
-        estr = '' if len(e.args) < 1 else (': ' + str(e))
-        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
-        return False
 
-    with shelve.open(urlmapfn) as url2hash:
+    See Also
+    --------
+    cache_contents : obtain a dictionary listing everything in the cache
+    """
+    with _cache() as (dldir, url2hash):
         return list(url2hash.keys())
+
+
+def cache_contents():
+    """Obtain a dict mapping cached URLs to filenames."""
+    with _cache() as (dldir, url2hash):
+        return ReadOnlyDict((k, os.path.join(dldir, v))
+                            for (k, v) in url2hash.items())
+
+
+_cache_zip_index_name = "index.json"
+
+
+def export_download_cache(filename_or_obj, urls=None):
+    """Exports the cache contents as a ZIP file
+
+    Parameters
+    ----------
+    filename_or_obj : str or file-like
+        Where to put the created ZIP file. Must be something the zipfile
+        module can write to.
+    urls : iterable of str or None
```

Comment:
This means just the "true" URLs, right, not the ones in `sources` too?

Reply:
Right. `sources` is just detail of how you got the file; it is notionally the contents of the true URL, and more relevantly, you ask for it by querying the true URL. I'll clarify.